celery/backends/new_cassandra.py

```diff
@@ -0,0 +1,199 @@
+# -* coding: utf-8 -*-
+"""
+    celery.backends.new_cassandra
+    ~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Apache Cassandra result store backend using DataStax driver
+
+"""
+from __future__ import absolute_import
+
+import sys
+import six
+try:  # pragma: no cover
+    import cassandra
+except ImportError:  # pragma: no cover
+    cassandra = None   # noqa
+
+from celery import states
+from celery.exceptions import ImproperlyConfigured
+from celery.utils.log import get_logger
+from .base import BaseBackend
+
+__all__ = ['NewCassandraBackend']
+
+logger = get_logger(__name__)
+
+
+class NewCassandraBackend(BaseBackend):
+    """New Cassandra backend utilizing DataStax driver
+
+    .. attribute:: servers
+
+        List of Cassandra servers with format: ``hostname``
+
+    :raises celery.exceptions.ImproperlyConfigured: if
+        module :mod:`cassandra` is not available.
+
+    """
+    servers = []
+    keyspace = None
+    table = None
+    supports_autoexpire = True      # autoexpire supported via entry_ttl
+
+    def __init__(self, servers=None, keyspace=None, table=None, entry_ttl=None,
+                 port=9042, **kwargs):
+        """Initialize Cassandra backend.
+
+        Raises :class:`celery.exceptions.ImproperlyConfigured` if
+        the :setting:`CASSANDRA_SERVERS` setting is not set.
+
+        """
+        super(NewCassandraBackend, self).__init__(**kwargs)
+
+        if not cassandra:
+            raise ImproperlyConfigured(
+                'You need to install the cassandra library to use the '
+                'Cassandra backend. See https://github.com/datastax/python-driver')
+
+        conf = self.app.conf
+        self.servers = (servers or
+                        conf.get('CASSANDRA_SERVERS') or
+                        self.servers)
+        self.port = (port or
+                     conf.get('CASSANDRA_PORT'))
+        self.keyspace = (keyspace or
+                         conf.get('CASSANDRA_KEYSPACE') or
+                         self.keyspace)
+        self.table = (table or
+                      conf.get('CASSANDRA_TABLE') or
+                      self.table)
+        expires = (entry_ttl or conf.get('CASSANDRA_ENTRY_TTL', None))
+
+        if expires is not None:
+            self.cqlexpires = ' USING TTL %s' % (expires, )
+        else:
+            self.cqlexpires = ''
+
+        read_cons = conf.get('CASSANDRA_READ_CONSISTENCY') or 'LOCAL_QUORUM'
+        write_cons = conf.get('CASSANDRA_WRITE_CONSISTENCY') or 'LOCAL_QUORUM'
+        try:
+            self.read_consistency = getattr(cassandra.ConsistencyLevel,
```

Comment:
Specify default value for `getattr` instead of catching `AttributeError`, same for the lines below